ModelViewer/ModelViewer/ViewerGraphicsWindow.cpp

```diff
@@ -57,6 +57,136 @@ bool ViewerGraphicsWindow::loadModel(QString filepath) {
     return m_currentModel.m_isValid;
 }
 
+bool ViewerGraphicsWindow::loadVertexShader(QString vertfilepath)
+{
+    if (!initialized)
+    {
+        return false;
+    }
+    if (vertfilepath.isEmpty()) {
+        vertfilepath = QFileDialog::getOpenFileName(nullptr, "Load Vertex Shader", "../Data/Shaders/", "");
+        if (vertfilepath.isEmpty()) {
+            return false;
+        }
+    }
+
+    //QOpenGLShaderProgram *newProgram = new QOpenGLShaderProgram(this);
+    m_program->removeAllShaders();
+
+    m_program->addShaderFromSourceFile(QOpenGLShader::Vertex, vertfilepath);
```

Comment:
Wherever we call addShaderFromSourceFile(), we need to check the return value.  The return value indicates if compilation was successful or not.

Reply:
Okay. I just decided to stop checking for compilation errors, because loading in one shader at a time breaks you code more ofte than not, and we needed to remember the shader even if it breaks your code because it might work once you load another shader.